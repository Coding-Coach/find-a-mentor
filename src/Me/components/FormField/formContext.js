import React from 'react';

export const formFieldContext = React.createContext();
