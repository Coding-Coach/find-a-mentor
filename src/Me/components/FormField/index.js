export { FormField as default } from './FormField';
