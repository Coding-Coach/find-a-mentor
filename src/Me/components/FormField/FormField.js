import React, { useState } from 'react';
import styled from 'styled-components';
import uniqueId from 'lodash/uniqueId';
import { formFieldContext } from './formContext';

const FormFieldContainer = styled.div`
  display: flex;
  flex-direction: column;
  font-family: Lato, sans-serif;
  margin-bottom: 30px;
`;
const Label = styled.label`
  font-weight: bold;
  font-size: 14px;
  line-height: 17px;
  margin-bottom: 6px;
`;

export const FormField = ({ label, className, children }) => {
  const [id] = useState(() => uniqueId('form-field-'));
  return (
    <FormFieldContainer className={className}>
      {label && <Label htmlFor={id}>{label}</Label>}
      <formFieldContext.Provider value={id}>
        {children}
      </formFieldContext.Provider>
    </FormFieldContainer>
  );
};
