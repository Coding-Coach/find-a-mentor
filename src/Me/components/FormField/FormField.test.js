import React from 'react';
import FormField from './';
import { render } from '@testing-library/react';
import { formFieldContext } from './formContext';

describe('FormField component', () => {
  it('renders properly', () => {
    const { container } = render(<FormField label="test label" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders the label when it is defined', () => {
    const { getByText } = render(<FormField label="test label" />);
    expect(getByText('test label')).toBeInTheDocument();
  });
  it('passes down the id', () => {
    let formFieldId;
    render(
      <FormField>
        <formFieldContext.Consumer>
          {id => (formFieldId = id)}
        </formFieldContext.Consumer>
      </FormField>
    );
    expect(formFieldId).toMatch('form-field-');
  });
});
