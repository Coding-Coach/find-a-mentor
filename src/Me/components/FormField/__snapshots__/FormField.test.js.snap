// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField component renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: Lato,sans-serif;
  margin-bottom: 30px;
}

.c1 {
  font-weight: bold;
  font-size: 14px;
  line-height: 17px;
  margin-bottom: 6px;
}

<div
  class="c0"
>
  <label
    class="c1"
    for="form-field-1"
  >
    test label
  </label>
</div>
`;
